### PR TITLE
Add readme

### LINKED ISSUE #8: Add a readme file

### PR DESCRIPTION
**Pull request by nicolaserny**
Add some documentation for the vscode store

fixes #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,13 +2,24 @@
...
 [![Build Status](https://travis-ci.org/xcomponent/vscode-xcomponent.svg?branch=master)](https://travis-ci.org/xcomponent/vscode-xcomponent)
 
-## Visual Studio Code extension for XComponent
-This repository contains the code of:
-* the vscode extension
-* the component viewer used in the extension
+# Visual Studio Code extension for [XComponent](http://www.xcomponent.com/)
+
+This repository contains:
+* [xcomponent extension](https://github.com/xcomponent/vscode-xcomponent/blob/master/extension/README.md)
```
Comment (maurelio1234) [inline README.md]: not sure this is useful. need a one liner on what can we do with each package

File: extension/README.md [hunk 2]
```diff
@@ -1,2 +1,34 @@
-# vscode-xcomponent
-visual studio code extension for xcomponent
+# xcomponent
+
+This extension provides a development environment for [XComponent](http://www.xcomponent.com/).
```
Comment (maurelio1234) [inline extension/README.md]: this should be on the higher level :)

on this level we need a more low level explanation (low level != longer)

File: extension/README.md [hunk 3]
```diff
@@ -1,2 +1,34 @@
...
+
+XComponent is a platform to easily create, monitor and share **microservices**. To get more details about XComponent, go to [the resource center](https://github.com/xcomponent/xcomponent).
+
+The extension includes the following features:
+* cxml preview
```
Comment (maurelio1234) [inline extension/README.md]: let's use the same naming convention everywhere: it's a component preview.

File: extension/README.md [hunk 4]
```diff
@@ -1,2 +1,34 @@
...
+
+More features yet to come...
+
+## Feature details
+
+## cxml preview
+
+You can display a graphical preview (state machine diagram) of a cxml file by using the *"XComponent: component preview"* command.
```
Comment (maurelio1234) [inline extension/README.md]: To display a component *.cxml file use the *"XComponent: Preview Component"* command.

File: viewer/README.md [hunk 5]
```diff
@@ -1,2 +1,7 @@
 # XComponent viewer
+
 Web viewer used in xcomponent visual studio code extension
```
Comment (maurelio1234) [inline viewer/README.md]: XComponent Visual Studio Code extension

File: README.md [hunk 6]
```diff
@@ -2,13 +2,24 @@
...
+* [xcomponent viewer](https://github.com/xcomponent/vscode-xcomponent/tree/master/viewer) used in the extension
```
Comment (maurelio1234) [inline README.md]: viewer for `*.cxml` files used in the extension

File: extension/README.md [hunk 7]
```diff
@@ -1,2 +1,34 @@
-# vscode-xcomponent
-visual studio code extension for xcomponent
+# xcomponent
```
Comment (maurelio1234) [inline extension/README.md]: XComponent

File: extension/README.md [hunk 8]
```diff
@@ -1,2 +1,34 @@
...
+XComponent is a platform to easily create, monitor and share **microservices**. To get more details about XComponent, go to [the resource center](https://github.com/xcomponent/xcomponent).
+
+The extension includes the following features:
+* cxml preview
+
+More features yet to come...
+
+## Feature details
+
+## cxml preview
```
Comment (maurelio1234) [inline extension/README.md]: Component preview

File: extension/README.md [hunk 9]
```diff
@@ -1,2 +1,34 @@
...
+
+You can display a graphical preview (state machine diagram) of a cxml file by using the *"XComponent: component preview"* command.
+
+![cxml preview](https://raw.githubusercontent.com/xcomponent/vscode-xcomponent/master/extension/images/cxml_preview.gif)
```
Comment (maurelio1234) [inline extension/README.md]: Component preview

File: README.md (hunk 1 above)
Comment (nicolaserny) [inline README.md]: I change the introduction.
As we talked about yesterday, I have the same introduction on the main readme et the extension readme (used by vscode store).

File: extension/README.md (hunk 7 above)
Comment (nicolaserny) [inline extension/README.md]: Done

File: extension/README.md (hunk 2 above)
Comment (nicolaserny) [inline extension/README.md]: I changed this part. Hope it's better...

File: extension/README.md (hunk 3 above)
Comment (nicolaserny) [inline extension/README.md]: done

File: extension/README.md (hunk 8 above)
Comment (nicolaserny) [inline extension/README.md]: done

File: extension/README.md (hunk 4 above)
Comment (nicolaserny) [inline extension/README.md]: Done

File: extension/README.md (hunk 9 above)
Comment (nicolaserny) [inline extension/README.md]: Done

File: viewer/README.md (hunk 5 above)
Comment (nicolaserny) [inline viewer/README.md]: I replaced by a simple comment.
The readme file of the viewer is not really the main purpose of this pull request (compared to the other readme)

**Comment by nicolaserny** [comment]
- Readme at the root of the project: it's much more interesting for developers. It contains a general description, a link to the getting started and details for developers
- Readme in the extension folder: it is used by the vscode store as a description. So it targets end users (basically how to use the extension).

**Pull request merged**